zcash_client_sqlite/src/wallet/init/migrations/orchard_shardtree.rs

```diff
@@ -0,0 +1,214 @@
+            // init height.
+            if let Some((start, end, range_priority)) = transaction
+                .query_row_and_then(
+                    "SELECT block_range_start, block_range_end, priority
+                     FROM scan_queue
+                     WHERE block_range_start <= :orchard_init_height
+                     AND block_range_end > :orchard_init_height",
+                    named_params![":orchard_init_height": u32::from(orchard_init_height)],
+                    |row| {
+                        let start = BlockHeight::from(row.get::<_, u32>(0)?);
+                        let end = BlockHeight::from(row.get::<_, u32>(1)?);
+                        let range_priority: i64 = row.get(2)?;
+                        Ok((start, end, range_priority))
+                    },
+                )
+                .optional()?
+            {
+                transaction.execute(
+                    "DELETE from scan_queue WHERE block_range_start = :start",
+                    named_params![":start": u32::from(start)],
+                )?;
+                // Rewrite the start of the scan range to be exactly what it was prior to the
+                // change.
+                transaction.execute(
+                    "INSERT INTO scan_queue (block_range_start, block_range_end, priority)
+                     VALUES (:block_range_start, :block_range_end, :priority)",
+                    named_params![
+                        ":block_range_start": u32::from(start),
+                        ":block_range_end": u32::from(orchard_init_height),
+                        ":priority": range_priority,
+                    ],
+                )?;
+                // Rewrite the remainder of the range to have priority `Historic`
+                transaction.execute(
+                    "INSERT INTO scan_queue (block_range_start, block_range_end, priority)
+                     VALUES (:block_range_start, :block_range_end, :priority)",
+                    named_params![
+                        ":block_range_start": u32::from(orchard_init_height),
+                        ":block_range_end": u32::from(end),
+                        ":priority": priority_code(&ScanPriority::Historic),
+                    ],
+                )?;
+                // Rewrite any scanned ranges above the end of the first Orchard
+                // range to have priority `Historic`
+                transaction.execute(
+                    "UPDATE scan_queue SET priority = :historic
+                     WHERE :block_range_start >= :orchard_initial_range_end
+                     AND priority = :scanned",
+                    named_params![
+                        ":historic": priority_code(&ScanPriority::Historic),
+                        ":orchard_initial_range_end": u32::from(end),
+                        ":scanned": priority_code(&ScanPriority::Scanned),
```

Comment:
Is it okay to only bump `Scanned` ranges, and not `Ignored` ranges? I think so, because in this migration we are assuming that the "Orchard birthday" is after the wallet birthday, and IIRC `Ignored` ranges are only created between Sapling activation and the wallet birthday.

Reply:
For future-proofing, I encoded this as `priority < :historic` (as the priority codes are by design ordered).